express-notes-api/server.js

```diff
@@ -0,0 +1,98 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.get('/api/notes', async (req, res) => {
+  const notesData = JSON.parse(await readFile('./data.json', { encoding: 'utf8' }));
+  const notes = notesData.notes;
+  const notesArr = [];
+  for (const note in notes) {
+    notesArr.push(notes[note]);
+  }
+  res.json(notesArr);
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const notesData = JSON.parse(await readFile('./data.json', { encoding: 'utf8' }));
+  const notes = notesData.notes;
+  console.log(typeof req.params.id);
+  if (Number.isInteger(parseInt(req.params.id)) || parseInt(req.params.id) < 1 || !parseInt(req.params.id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
```

Comment:
Please double-check that the server responds with `400 Bad Request`s for the right values of `id`:

---

![image](https://github.com/JCJenvey/m2-0323-code-solutions/assets/16143771/960e3170-8a0b-4cf7-8845-6795d277e9bc)